### DISCUSSION TITLE
Absolute Imports and Module path aliases

### DISCUSSION THREAD

**Discussion by hajpepe** [vercel/next.js#38147] [Help] [answered]
hello dear , 
guys my project is not ts and by deafult dont have jsconfig.json
i created like this

```
// tsconfig.json or jsconfig.json
{
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@/components/*": ["components/*"]
    }
  }
}
```
but @ not working for my project
i need do somthing ? like verify jsconfig in next.config ?

**Comment by icyJoseph** [comment, marked as answer]
Hi,

I tried this and it worked, exactly with your configuration.

```js
import { Button } from "@/components/Button";
```

Could you share the repository where you are doing this?

I used create next-app, and gave the project a random name:

```
 yarn create next-app test-absolute-paths-js
```

Are you, perhaps using the `src` directory?

**Reply by hajpepe** [reply]
no i dont use `src `dir

**Reply by hajpepe** [reply]
my components and constants in page folder 
![image](https://user-images.githubusercontent.com/58140037/176444888-38b8b182-e4ef-40fd-a250-b57324309deb.png)

**Reply by icyJoseph** [reply]
Aha, you shouldn't do that.

Everything inside pages is literally a page, so people would be able to see `/pages/components/button` in the browser.

Please use this structure:

```
.
├── README.md
├── components
├── jsconfig.json
├── next.config.js
├── node_modules
├── package.json
├── pages
├── public
├── styles
└── yarn.lock
```

With `components` at the same level as `pages`. Same for constants and such, this is why some people prefer the `src` directory, which groups `components`, `pages`, `constants`, etc...

**Reply by akshayvarma72** [reply]
@icyJoseph  is it possible to put `public` folder in `src` as well ?

**Reply by icyJoseph** [reply]
Nope, https://nextjs.org/docs/advanced-features/src-directory#caveats:

> Config files like next.config.js and tsconfig.json, as well as environment variables, should be inside the root directory, moving them to src won't work. Same goes for the [public directory](https://nextjs.org/docs/basic-features/static-file-serving)

**Comment by ThisisMS11** [comment]
I am also facing the same issue with the same jsconfig configuration , it sometimes work and also show suggestions for components directory present in the root directory but when you stop and restart your development server it seems to have forget the component folder path . this is very weird to find in nextjs 13 app router. because of this  i still use the traiditional way of importing components to be on the safer side while production .
